linter.go

```diff
@@ -0,0 +1,13 @@
+// +build linter
+
+package main
+
+import _ "github.com/3rf/mongo-lint/golint"
```

Comment:
I think this file is necessary to retain a vendored third-party tool that we use in Evergreen. Without this file, `go mod tidy` will remove it.

Some more explanation [here](https://github.com/golang/go/wiki/Modules#how-can-i-track-tool-dependencies-for-a-module) on why it seems to be the best way to do it.

Reply:
Seems reasonable since it's following the golang convention.